packages/data-point/lib/reducer/resolve.js

```diff
@@ -41,17 +46,16 @@ module.exports.getReducerFunction = getReducerFunction
 /**
  * apply a Reducer to an accumulator
  * @param {Object} manager
- * @param {Function} resolveTransform
  * @param {Accumulator} accumulator
  * @param {Reducer} reducer
  * @returns {Promise<Accumulator>}
  */
-function resolve (manager, resolveTransform, accumulator, reducer) {
-  const reducerFunction = getReducerFunction(
-    manager,
-    resolveTransform,
-    reducer.type
-  )
+function resolve (manager, accumulator, reducer) {
+  if (!reducer) {
+    return Promise.resolve(accumulator)
+  }
```

Comment:
this conditional is here because the entity resolve function does not check if lifecycle functions are defined before trying to resolve them:

https://github.com/raingerber/data-point/blob/rename-transform-expression/packages/data-point/lib/entity-types/base-entity/resolve.js#L140

Reply:
Is that not these two checks?

https://github.com/ViacomInc/data-point/blob/825f9a3b9c1cc7dd2f627801e9411718cbc91e31/packages/data-point/lib/entity-types/base-entity/resolve.js#L13

https://github.com/ViacomInc/data-point/blob/825f9a3b9c1cc7dd2f627801e9411718cbc91e31/packages/data-point/lib/transform-expression/resolve.js#L16

Reply:
the code in the first link has been refactored to use `utils.reducerIsEmpty(errorReducer)`, and the second file you're linking to doesn't exist on this branch, which is why I added this code to replace it.

Reply:
OK, thank you for the original comment and the response.